plugins/producer_plugin/include/eosio/producer_plugin/unapplied_transaction_queue.hpp

```diff
@@ -0,0 +1,109 @@
+/**
+ *  @file
+ *  @copyright defined in eos/LICENSE
+ */
+
+#pragma once
+
+#include <eosio/chain/transaction_metadata.hpp>
+#include <eosio/chain/block_state.hpp>
+
+#include <queue>
+
+namespace eosio {
+
+/**
+ * Track unapplied transactions for aborted blocks, forked blocks, and subjectively
+ * failed transactions.
+ */
+class unapplied_transaction_queue {
+   std::deque<std::vector<chain::transaction_metadata_ptr>> aborted_block_trxs;
+   size_t current_trx_in_trxs = 0;
+   std::deque<chain::branch_type> forked_branches;
+   size_t current_trx_in_block = 0;
+   std::deque<chain::transaction_metadata_ptr> subjective_failed_trxs;
+public:
+
+   bool empty() const {
+      return aborted_block_trxs.empty() && forked_branches.empty() && subjective_failed_trxs.empty();
+   }
+
+   void clear() {
+      aborted_block_trxs.clear();
+      current_trx_in_trxs = 0;
+      forked_branches.clear();
+      current_trx_in_block = 0;
+      subjective_failed_trxs.clear();
+   }
+
+   void add_aborted( std::vector<chain::transaction_metadata_ptr> aborted_trxs ) {
+      if( aborted_trxs.empty() ) return;
+      // fifo queue so pop back
+      aborted_block_trxs.emplace_front( std::move( aborted_trxs ) );
+   }
+
+   void add_subjective_failure( chain::transaction_metadata_ptr trx ) {
+      // fifo queue so pop back
+      subjective_failed_trxs.emplace_front( std::move( trx ) );
+   }
+
+   void add_forked( chain::branch_type forked_branch ) {
+      if( forked_branch.empty() ) return;
+      // forked_branch is in reverse order
+      // fifo queue for branches so pop back
+      forked_branches.emplace_front( std::move( forked_branch ) );
+   }
+
+   /// Order returned: aborted block transactions, subjectively failed, forked
```

Comment:
Shouldn't the order be forked, aborted block transactions, and then subjectively failed?

Reply:
Changed to: forked, aborted, subjective